### PR TITLE
how to use blazor wasm with odata client

### PR DESCRIPTION
**Pull request by ElizabethOkerio**

### CODE REVIEW

File: Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+Start by creating a Blazor WebAssembly project.
+
+Open Visual Studio and click on Create a new project then select the Blazor App template and click Next. Configure your project by providing the project name and the location path then click Create. On the Create Project Page that appears, specify the target framework that you want to use and select Blazor WebAssembly App then click Create. A blazor web assembly project will be created like below. 
+
+![SAMPLE BLAZOR WEB ASSEMBLY PROJECT](../../assets/2021-02-02-blazor-wasm-app-sample.png)
+
+There are two ways in which you can use OData Client with blazor web assembly: 
+
+* You can use the Microsoft OData Connected Service
+* You can POCO's
```
Comment (habbes) [inline Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md:24]: Maybe you could expand the abbreviation here, or say "You can use plain-old C# objects (POCOs)"

File: Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+ms.topic: article
+ 
+---
+# How To Use Blazor Web Assembly with OData Client 
+**Applies To**: [!INCLUDE[appliesto-odataclient](../../includes/appliesto-odataclient-v6.md)]
+
+# Getting Started
+Start by creating a Blazor WebAssembly project.
+
+Open Visual Studio and click on Create a new project then select the Blazor App template and click Next. Configure your project by providing the project name and the location path then click Create. On the Create Project Page that appears, specify the target framework that you want to use and select Blazor WebAssembly App then click Create. A blazor web assembly project will be created like below.
```
Comment (habbes) [inline Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md:17]: I think you should make the UI texts stand out, maybe put them in quotes like "Create a new project" or make them bold?

File: Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+Uri serviceRoot = new Uri("https://localhost:44353/odata");
+var dataserviceContect = new Default.Container(serviceRoot);
+dataserviceContect.HttpRequestTransportMode = HttpRequestTransportMode.HttpClient;
+var result = await dataserviceContect.Books.ExecuteAsync();
+```
+You can then display the data however you want on your razor pages.  
+
+## Using POCO's
+
+If you do not wish to use Microsoft Odata Connected Service or any code generation tools, then you create the `DataServiceContext` like below:
```
Comment (habbes) [inline Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md:45]:
```suggestion
If you do not wish to use Microsoft OData Connected Service or any code generation tools, then you create the `DataServiceContext` like below: 
```

File: Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+```
+
+
+
+
+
+
+
+
+
```
Comment (habbes) [inline Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md:92]: Remove the extra lines?

File: Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+var result = await dataserviceContect.Books.ExecuteAsync();
+```
+You can then display the data however you want on your razor pages.  
+
+## Using POCO's
+
+If you do not wish to use Microsoft Odata Connected Service or any code generation tools, then you create the `DataServiceContext` like below: 
+
+```c#
+public partial class ODataServiceContext : DataServiceContext
```
Comment (habbes) [inline Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md:48]: Why is this a partial class?

File: Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+    public Connector(Uri serviceRoot) : base(serviceRoot)
+    {
+        HttpRequestTransportMode = HttpRequestTransportMode.HttpClient;
+
+        this.Books = base.CreateQuery<Book>("Books");
+        this.Format.LoadServiceModel = () => GetEDMModel();
+        this.Format.UseJson();
+    }
+
+    private IEdmModel GetEDMModel()
```
Comment (habbes) [inline Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md:59]:
```suggestion
    private IEdmModel GetEdmModel()
```

File: Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+
+```c#
+public partial class ODataServiceContext : DataServiceContext
+{
+    public Connector(Uri serviceRoot) : base(serviceRoot)
+    {
+        HttpRequestTransportMode = HttpRequestTransportMode.HttpClient;
+
+        this.Books = base.CreateQuery<Book>("Books");
+        this.Format.LoadServiceModel = () => GetEDMModel();
```
Comment (habbes) [inline Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md:55]:
```suggestion
        this.Format.LoadServiceModel = () => GetEdmModel();
```

File: Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+description: ""
+
+author: elokerio
+ms.author: elokerio
+ms.date: 1/18/2021
+ms.topic: article
+ 
+---
+# How To Use Blazor Web Assembly with OData Client 
+**Applies To**: [!INCLUDE[appliesto-odataclient](../../includes/appliesto-odataclient-v6.md)]
```
Comment (KenitoInc) [inline Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md:12]: A few things to note:
`author` - should be your github username.

I update the docs to v7 
`**Applies To**: [!INCLUDE[appliesto-odataclient](../../includes/appliesto-odataclient-v7.md)]`

The file should be in `/Odata-docs/client/using_blazor_wasm_with_odata_client.md`

File: Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md [hunk 9]
```diff
@@ -0,0 +1,92 @@
...
+
+Then on the `OnInitializedAsync` method in any of the razor pages add the code below to get data from an endpoint for example.
+```c# 
+var dataserviceContect = new Connector(new Uri("https://localhost:44353/odata"));
+var result = await dataserviceContect.Books.ExecuteAsync();
+
+foreach (var book in result)
+{
+    Console.WriteLine(book.Author);
+    Console.WriteLine(book.ISBN);
```
Comment (habbes) [inline Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md:80]: Out of curiosity, will this output the info on the browser console?

**Review by KenitoInc — CHANGES_REQUESTED** [review]
You need to adding a reference to this file in the Table of Contents https://github.com/MicrosoftDocs/OData-docs/blob/15acf536ef0ee7ab2a68050e2999653c3c5a6b70/Odata-docs/TOC.yml#L277

File: Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md (hunk 1 above)
Comment (ElizabethOkerio) [reply Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md]: fixed

File: Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md (hunk 6 above)
Comment (ElizabethOkerio) [reply Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md]: resolved

File: Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md (hunk 3 above)
Comment (ElizabethOkerio) [reply Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md]: resolved

File: Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md (hunk 1 above)
Comment (ElizabethOkerio) [reply Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md]: resolved

File: Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md (hunk 5 above)
Comment (ElizabethOkerio) [reply Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md]: OData connected service usually generated a partial class. Just for consistency.

File: Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md (hunk 8 above)
Comment (ElizabethOkerio) [reply Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md]: resolved

File: Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md (hunk 4 above)
Comment (ElizabethOkerio) [reply Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md]: resolved

File: Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md (hunk 2 above)
Comment (ElizabethOkerio) [reply Odata-docs/client/v6/using_blazor_wasm_with_odata_client.md]: fixed

**Review by KenitoInc — APPROVED** [review]
LGTM

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+title: "Blazor Wasm with OData Client"
+description: "How to use blazor with odata client"
+
+author: ElizabethOkerio
+ms.author: elokerio
+ms.date: 1/18/2021
+ms.topic: article
+ 
+---
+# How To Use Blazor Web Assembly with OData Client
```
Comment (wandeg) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:11]:
```suggestion
# How To Use Blazor WebAssembly with OData Client 
```

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+**Applies To**: [!INCLUDE[appliesto-odataclient](../includes/appliesto-odataclient-v7.md)]
+
+## Getting Started
+Start by creating a Blazor WebAssembly project.
+
+Open Visual Studio and click on "Create a new project" then select the "Blazor App template" and click "Next". Configure your project by providing the project name and the location path then click "Create". On the "Create Project" Page that appears, specify the target framework that you want to use and select "Blazor WebAssembly App" then click "Create". A blazor web assembly project will be created like below. 
+
+![SAMPLE BLAZOR WEB ASSEMBLY PROJECT](../assets/2021-02-02-blazor-wasm-app-sample.png)
+
+There are two ways in which you can use OData Client with blazor web assembly:
```
Comment (wandeg) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:21]:
```suggestion
There are two ways in which you can use OData Client with Blazor WebAssembly: 
```

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+![SAMPLE BLAZOR WEB ASSEMBLY PROJECT](../assets/2021-02-02-blazor-wasm-app-sample.png)
+
+There are two ways in which you can use OData Client with blazor web assembly: 
+
+* You can use the Microsoft OData Connected Service
+* You can use plain-old C# objects (POCOs)
+
+## Using Microsoft OData Connected Service
+
+Add OData Connected service to the project you just created above.
```
Comment (wandeg) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:28]:
```suggestion
Add OData Connected Service to the project you just created above. 
```

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 13]
```diff
@@ -0,0 +1,83 @@
...
+* You can use plain-old C# objects (POCOs)
+
+## Using Microsoft OData Connected Service
+
+Add OData Connected service to the project you just created above. 
+
+To do so, ensure that you have the Microsoft OData Connected Service extension installed. Refer to this link [Microsoft OData Connected Service](/odata/connectedservice/getting-started/) on how to get started with Microsoft OData Connected Service.
+Right click on your project and add a connected service. Select the Microsoft OData Connected Service that you installed and use the Wizard to generate proxy classes for the OData endpoint that you want to use.
+
+Once all this is done, Open any of the Blazor razor pages that you want to display data from an OData endpoint on and include the code below on the `OnInitializedAsync` method.
```
Comment (wandeg) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:33]:
```suggestion
Once all this is done, open any of the Blazor Razor pages from which you want to display data from an OData endpoint and include the code below in the `OnInitializedAsync` method.  
```

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 14]
```diff
@@ -0,0 +1,83 @@
...
+
+Once all this is done, Open any of the Blazor razor pages that you want to display data from an OData endpoint on and include the code below on the `OnInitializedAsync` method.  
+
+```c#
+Uri serviceRoot = new Uri("https://localhost:44353/odata");
+var dataserviceContect = new Default.Container(serviceRoot);
+dataserviceContect.HttpRequestTransportMode = HttpRequestTransportMode.HttpClient;
+var result = await dataserviceContect.Books.ExecuteAsync();
+```
+You can then display the data however you want on your razor pages.
```
Comment (wandeg) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:41]:
```suggestion
You can then display the data however you want on your Razor pages.  
```

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+        builder.EntitySet<Book>("Books").EntityType.HasKey(p => p.Id).Count().Select().Page(null, 100).Expand().Filter();
+
+        return builder.GetEdmModel();
+    }
+
+    public DataServiceQuery<Book> Books { get; }
+}
+```
+
+Then on the `OnInitializedAsync` method in any of the razor pages add the code below to get data from an endpoint for example.
```
Comment (wandeg) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:72]:
```suggestion
Then in the `OnInitializedAsync` method in any of the Razor pages add the code below to get data from an endpoint. For example:
```

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 16]
```diff
@@ -0,0 +1,83 @@
...
+ms.topic: article
+ 
+---
+# How To Use Blazor Web Assembly with OData Client 
+**Applies To**: [!INCLUDE[appliesto-odataclient](../includes/appliesto-odataclient-v7.md)]
+
+## Getting Started
+Start by creating a Blazor WebAssembly project.
+
+Open Visual Studio and click on "Create a new project" then select the "Blazor App template" and click "Next". Configure your project by providing the project name and the location path then click "Create". On the "Create Project" Page that appears, specify the target framework that you want to use and select "Blazor WebAssembly App" then click "Create". A blazor web assembly project will be created like below.
```
Comment (wandeg) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:17]:
```suggestion
Open Visual Studio and click on "Create a new project" then select the "Blazor App template" and click "Next". Configure your project by providing the project name and the location path then click "Create". On the "Create Project" Page that appears, specify the target framework that you want to use and select "Blazor WebAssembly App" then click "Create". A Blazor WebAssembly project will be created like below. 
```

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 17]
```diff
@@ -0,0 +1,83 @@
...
+
+* You can use the Microsoft OData Connected Service
+* You can use plain-old C# objects (POCOs)
+
+## Using Microsoft OData Connected Service
+
+Add OData Connected service to the project you just created above. 
+
+To do so, ensure that you have the Microsoft OData Connected Service extension installed. Refer to this link [Microsoft OData Connected Service](/odata/connectedservice/getting-started/) on how to get started with Microsoft OData Connected Service.
+Right click on your project and add a connected service. Select the Microsoft OData Connected Service that you installed and use the Wizard to generate proxy classes for the OData endpoint that you want to use.
```
Comment (wandeg) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:31]:
```suggestion
Right click on your project and add a Connected Service. Select the Microsoft OData Connected Service that you installed and use the Wizard to generate proxy classes for the OData endpoint that you want to use.
```

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md (hunk 10 above)
Comment (ElizabethOkerio) [reply Odata-docs/client/using-blazor-wasm-with-odata-client.md:11]: fixed

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md (hunk 11 above)
Comment (ElizabethOkerio) [reply Odata-docs/client/using-blazor-wasm-with-odata-client.md:21]: fixed

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md (hunk 12 above)
Comment (ElizabethOkerio) [reply Odata-docs/client/using-blazor-wasm-with-odata-client.md:28]: fixed

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md (hunk 16 above)
Comment (gathogojr) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:17]: My suggestion would be to convert this paragraph into an ordered or unordered list so it's easier to follow

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md (hunk 13 above)
Comment (ElizabethOkerio) [reply Odata-docs/client/using-blazor-wasm-with-odata-client.md:33]: I don't think you can use from which here.. maybe on which

Comment (habbes) [reply Odata-docs/client/using-blazor-wasm-with-odata-client.md:33]: What about: open any of the Blazor razor pages where data from the OData endpoint should be displayed. Then, include the below in the `OnInitializeAsync` method.

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 18]
```diff
@@ -0,0 +1,83 @@
...
+var result = await dataserviceContect.Books.ExecuteAsync();
+```
+You can then display the data however you want on your razor pages.  
+
+## Using plain-old C# objects (POCOs)
+
+If you do not wish to use Microsoft Odata Connected Service or any code generation tools, then you create the `DataServiceContext` like below: 
+
+```c#
+public partial class ODataServiceContext : DataServiceContext
```
Comment (habbes) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:48]: I think it would be helpful to mention which namespace to include in order to get `DataServiceContext`.

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md (hunk 14 above)
Comment (ElizabethOkerio) [reply Odata-docs/client/using-blazor-wasm-with-odata-client.md:41]: fixed

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 19]
```diff
@@ -0,0 +1,83 @@
...
+There are two ways in which you can use OData Client with blazor web assembly: 
+
+* You can use the Microsoft OData Connected Service
+* You can use plain-old C# objects (POCOs)
+
+## Using Microsoft OData Connected Service
+
+Add OData Connected service to the project you just created above. 
+
+To do so, ensure that you have the Microsoft OData Connected Service extension installed. Refer to this link [Microsoft OData Connected Service](/odata/connectedservice/getting-started/) on how to get started with Microsoft OData Connected Service.
```
Comment (habbes) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:30]:
```suggestion
To do so, ensure that you have the Microsoft OData Connected Service extension installed. [Refer to this link](/odata/connectedservice/getting-started/) to find out how to get started with Microsoft OData Connected Service.
```

**Review by habbes — APPROVED** [review]
Minor changes requested, but other than that looks good to me.

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md (hunk 15 above)
Comment (ElizabethOkerio) [reply Odata-docs/client/using-blazor-wasm-with-odata-client.md:72]: solved

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md (hunk 16 above)
Comment (ElizabethOkerio) [reply Odata-docs/client/using-blazor-wasm-with-odata-client.md:17]: fixed

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 20]
```diff
@@ -0,0 +1,83 @@
...
+Right click on your project and add a connected service. Select the Microsoft OData Connected Service that you installed and use the Wizard to generate proxy classes for the OData endpoint that you want to use.
+
+Once all this is done, Open any of the Blazor razor pages that you want to display data from an OData endpoint on and include the code below on the `OnInitializedAsync` method.  
+
+```c#
+Uri serviceRoot = new Uri("https://localhost:44353/odata");
+var dataserviceContect = new Default.Container(serviceRoot);
+dataserviceContect.HttpRequestTransportMode = HttpRequestTransportMode.HttpClient;
+var result = await dataserviceContect.Books.ExecuteAsync();
```
Comment (habbes) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:39]:
```suggestion
var dataServiceContext = new Default.Container(serviceRoot);
dataServiceContext.HttpRequestTransportMode = HttpRequestTransportMode.HttpClient;
var result = await dataServiceContext.Books.ExecuteAsync();
```

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 21]
```diff
@@ -0,0 +1,83 @@
...
+    }
+
+    public DataServiceQuery<Book> Books { get; }
+}
+```
+
+Then on the `OnInitializedAsync` method in any of the razor pages add the code below to get data from an endpoint for example.
+```c# 
+var dataserviceContect = new Connector(new Uri("https://localhost:44353/odata"));
+var result = await dataserviceContect.Books.ExecuteAsync();
```
Comment (habbes) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:75]:
```suggestion
var dataServiceContext = new Connector(new Uri("https://localhost:44353/odata"));
var result = await dataServiceContext.Books.ExecuteAsync();
```

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md (hunk 17 above)
Comment (ElizabethOkerio) [reply Odata-docs/client/using-blazor-wasm-with-odata-client.md:31]: fixed

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 22]
```diff
@@ -0,0 +1,91 @@
...
+Uri serviceRoot = new Uri("https://localhost:44353/odata");
+var dataServiceContect = new Default.Container(serviceRoot);
+dataServiceContect.HttpRequestTransportMode = HttpRequestTransportMode.HttpClient;
+var result = await dataServiceContect.Books.ExecuteAsync();
+```
+You can then display the data however you want on your Razor pages.  
+
+## Using plain-old C# objects (POCOs)
+
+If you do not wish to use Microsoft Odata Connected Service or any code generation tools, then you create the `DataServiceContext` like below:
```
Comment (gathogojr) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:52]:
```suggestion
If you do not wish to use Microsoft OData Connected Service or any code generation tools, then you create the `DataServiceContext` like below: 
```

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 23]
```diff
@@ -0,0 +1,91 @@
...
+
+## Using plain-old C# objects (POCOs)
+
+If you do not wish to use Microsoft Odata Connected Service or any code generation tools, then you create the `DataServiceContext` like below: 
+The `DataServiceContext` class is found in the `Microsoft.OData.Client` namespace.
+
+```c#
+public partial class ODataServiceContext : DataServiceContext
+{
+    public Connector(Uri serviceRoot) : base(serviceRoot)
```
Comment (gathogojr) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:58]:
```suggestion
    public ODataServiceContext(Uri serviceRoot) : base(serviceRoot)
```

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 24]
```diff
@@ -0,0 +1,91 @@
...
+        return builder.GetEdmModel();
+    }
+
+    public DataServiceQuery<Book> Books { get; }
+}
+```
+
+Then in the `OnInitializedAsync` method in any of the Razor pages add the code below to get data from an endpoint. For example:
+```c# 
+var dataServiceContect = new Connector(new Uri("https://localhost:44353/odata"));
```
Comment (gathogojr) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:82]:
```suggestion
var dataServiceContect = new ODataServiceContext(new Uri("https://localhost:44353/odata"));
```

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 25]
```diff
@@ -0,0 +1,91 @@
...
+var result = await dataServiceContect.Books.ExecuteAsync();
+
+foreach (var book in result)
+{
+    Console.WriteLine(book.Author);
+    Console.WriteLine(book.ISBN);
+
```
Comment (gathogojr) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:89]:
```suggestion
```

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md (hunk 23 above)
Comment (ElizabethOkerio) [reply Odata-docs/client/using-blazor-wasm-with-odata-client.md]: fixed

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 26]
```diff
@@ -0,0 +1,91 @@
...
+        this.Books = base.CreateQuery<Book>("Books");
+        this.Format.LoadServiceModel = () => GetEdmModel();
+        this.Format.UseJson();
+    }
+
+    private IEdmModel GetEdmModel()
+    {
+        ODataConventionModelBuilder builder = new ODataConventionModelBuilder();
+        //builder.EntitySet
```
Comment (gathogojr) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:70]:
```suggestion

```

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 27]
```diff
@@ -0,0 +1,91 @@
...
+
+The use of the below syntax is possible with `Microsoft.OData.Client v7.8.1` and above. 
+
+```c#
+Uri serviceRoot = new Uri("https://localhost:44353/odata");
+var dataServiceContect = new Default.Container(serviceRoot);
+dataServiceContect.HttpRequestTransportMode = HttpRequestTransportMode.HttpClient;
+var result = await dataServiceContect.Books.ExecuteAsync();
+```
+You can then display the data however you want on your Razor pages.
```
Comment (gathogojr) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:48]: In the [section below](https://github.com/MicrosoftDocs/OData-docs/pull/169/files#diff-0295219f7f0d0c74c67044e3d2144690c7261a6bee0611c1b14be57eff79a424R85) you iterate through `result` and render `Author` and `ISBN`. Maybe you could do the same here...

**Review by gathogojr — APPROVED** [review]
Left a few comments otherwise LGTM. Great job!

nit: If in the **Getting Started** section, you instructed the user to create a specific Razor page (e.g. _Books.cshtml_), you'd be able to improve the following two lines so they don't feel mouthful - _"any of the Blazor Razor pages on which you want to display data..."_
- https://github.com/MicrosoftDocs/OData-docs/pull/169/files#diff-0295219f7f0d0c74c67044e3d2144690c7261a6bee0611c1b14be57eff79a424R38
- https://github.com/MicrosoftDocs/OData-docs/pull/169/files#diff-0295219f7f0d0c74c67044e3d2144690c7261a6bee0611c1b14be57eff79a424R78

**Review by wandeg — APPROVED** [review]

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 28]
```diff
@@ -0,0 +1,93 @@
...
+
+Once all this is done, open any of the Blazor Razor pages on which you want to display data from an OData endpoint and include the code below in the `OnInitializedAsync` method.  
+
+The use of the below syntax is possible with `Microsoft.OData.Client v7.8.1` and above. 
+
+```c#
+Uri serviceRoot = new Uri("https://localhost:44353/odata");
+var dataServiceContect = new Default.Container(serviceRoot);
+dataServiceContect.HttpRequestTransportMode = HttpRequestTransportMode.HttpClient;
+var result = await dataServiceContect.Books.ExecuteAsync();
```
Comment (habbes) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:46]:
```suggestion
Uri serviceRoot = new Uri("https://localhost:44353/odata");
var dataServiceContext = new Default.Container(serviceRoot);
dataServiceContext.HttpRequestTransportMode = HttpRequestTransportMode.HttpClient;
var result = await dataServiceContext.Books.ExecuteAsync();
```

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 29]
```diff
@@ -0,0 +1,93 @@
...
+    }
+
+    public DataServiceQuery<Book> Books { get; }
+}
+```
+
+Then in the `OnInitializedAsync` method in any of the Razor pages add the code below to get data from an endpoint. For example:
+```c# 
+var dataServiceContect = new ODataServiceContext(new Uri("https://localhost:44353/odata"));
+var result = await dataServiceContect.Books.ExecuteAsync();
```
Comment (habbes) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:86]:
```suggestion
var dataServiceContext = new ODataServiceContext(new Uri("https://localhost:44353/odata"));
var result = await dataServiceContext.Books.ExecuteAsync();
```

**Review by habbes — CHANGES_REQUESTED** [review]
Fix minor typos

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 30]
```diff
@@ -0,0 +1,93 @@
...
+Right click on your project and add a Connected Service. Select the Microsoft OData Connected Service that you installed and use the Wizard to generate proxy classes for the OData endpoint that you want to use.
+
+Once all this is done, open any of the Blazor Razor pages on which you want to display data from an OData endpoint and include the code below in the `OnInitializedAsync` method.  
+
+The use of the below syntax is possible with `Microsoft.OData.Client v7.8.1` and above. 
+
+```c#
+Uri serviceRoot = new Uri("https://localhost:44353/odata");
+var dataServiceContext = new Default.Container(serviceRoot);
+dataServiceContext.HttpRequestTransportMode = HttpRequestTransportMode.HttpClient;
```
Comment (odero) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:45]: Why is this necessary? Maybe add a comment

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 31]
```diff
@@ -0,0 +1,93 @@
...
+var result = await dataServiceContext.Books.ExecuteAsync();
+foreach (var book in result)
+{
+    Console.WriteLine(book.Author);
+    Console.WriteLine(book.ISBN);
+}
+```
+You can then display the data however you want on your Razor pages.  
+
+## Using plain-old C# objects (POCOs)
```
Comment (odero) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:55]:
```suggestion
## Using Plain Old CLR objects (POCOs)
```

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 32]
```diff
@@ -0,0 +1,93 @@
...
+* Finally on the "Create Project" page, specify the target framework that you want to use and select "Blazor WebAssembly App" then click "Create". 
+
+A Blazor Web Assembly project will be created like below. 
+
+![SAMPLE BLAZOR WEB ASSEMBLY PROJECT](../assets/2021-02-02-blazor-wasm-app-sample.png)
+
+There are two ways in which you can use OData Client with Blazor WebAssembly: 
+
+* You can use the Microsoft OData Connected Service
+* You can use plain-old C# objects (POCOs)
```
Comment (odero) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:29]:
```suggestion
* You can use Plain Old CLR Objects (POCOs)
```

File: Odata-docs/client/using-blazor-wasm-with-odata-client.md [hunk 33]
```diff
@@ -0,0 +1,93 @@
...
+{
+    Console.WriteLine(book.Author);
+    Console.WriteLine(book.ISBN);
+}
+```
+You can then display the data however you want on your Razor pages.  
+
+## Using plain-old C# objects (POCOs)
+
+If you do not wish to use Microsoft OData Connected Service or any code generation tools, then you create the `DataServiceContext` like the example below:
```
Comment (odero) [inline Odata-docs/client/using-blazor-wasm-with-odata-client.md:57]:
```suggestion
If you prefer to not use Microsoft OData Connected Service or any code generation tools, then you can create the `DataServiceContext` as in the example below:
```

**Review by odero — CHANGES_REQUESTED** [review]

**Review by odero — COMMENTED** [review]
Also worth going with John's feedback here https://github.com/MicrosoftDocs/OData-docs/pull/169#pullrequestreview-585476128

**Review by odero — APPROVED** [review]
